src/base/RefCount.h

```diff
@@ -26,6 +26,12 @@ template <class C>
 class RefCount
 {
 
+public:
+    template<typename... Args>
+    inline static auto Make(Args... args) {
+        return RefCount<C>(new C(args...));
```

Comment:
Aha, sweet. That is one new C++ feature I hadn't looked into uses for yet. Seems to work perfectly (pun intended).

Reply:
Not done AFAICT: This PR code is still copying (rather than perfectly forwarding) because it is still not using references. If the earlier wrapper() example was not helpful, perhaps this text about forwarding references will be: https://en.cppreference.com/w/cpp/language/reference#Forwarding_references

Please let me know if I should provide a specific code replacement instead. I just thought that showing where to insert a few magic symbols would jeopardize understanding of the underlying reasoning/mechanism (which is useful in other contexts as well).

Reply:
I have rebased since, it is not in the latest [24727e6](https://github.com/squid-cache/squid/pull/1458/commits/24727e6ad0d58d6c4788c6c14e8abc4612846744)

Reply:
Hmm, if I am understanding correctly you are suggesting to force the use of rvalue references instead of preserving the caller references (or lack of) ?
```c++
    inline static auto Make(Args&&... args) {
        return RefCount<C>(new C(std::forward<Args>(args)...));
    }
```

Reply:
> if I am understanding correctly you are suggesting to force the use of rvalue references instead of preserving the caller references (or lack of)?

I would not describe my suggestion as forcing the use of rvalue references. And following my suggestion would also preserve caller references, so that desirable preservation cannot be accurately described as "instead".

Perfect forwarding requires [forwarding references](https://en.cppreference.com/w/cpp/language/reference#Forwarding_references), which are not equivalent to rvalue references: Forwarding references use rvalue reference _declaration syntax_, but forwarding references effect/semantics is different from rvalue references. Not all references that use rvalue reference declaration syntax are forwarding references (because forwarding references require special/extra preconditions). For example, a classic move constructor uses an rvalue reference but not a forwarding reference. Due to those semantic differences, I believe that calling a forwarding reference an rvalue reference is also technically inaccurate, but since they share the same rvalue reference declaration syntax, I suspect that does happen a lot.

The updated branch code uses forwarding references, so I am resolving this change request, but if you would like to discuss any related concerns, I am happy to participate. Personally, I find this C++ concept difficult to master, and Herb Sutter [calls](https://stackoverflow.com/questions/29590883/herb-sutters-cppcon-perfect-forwarding-slides) it "unteachable", but since it is so useful, I believe that mastering it is worth the effort.

P.S. Forwarding references were previously known as "universal references". Some of the primary discussions about them can be found using that earlier term.